projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts

```diff
@@ -1211,10 +1211,18 @@ export class IgxDragDirective implements AfterContentInit, OnDestroy {
             originalEvent
         };
 
-        const elementsFromPoint = this.getElementsAtPoint(pageX, pageY);
+        let elementsFromPoint = this.getElementsAtPoint(pageX, pageY);
+
+        // Check for shadowRoot instance and use it if present
+        for (const elFromPoint of elementsFromPoint) {
```

Comment:
If there is a regular element without shadow root above the one with shadow root, would this for cycle skip that element above it?

Reply:
Thank you, Svetlio. @dobromirts lets create and app that will have a grid in iframe and a overlapping dialog that will consist two draggable/dropabble elements within. After the grid iframe is loaded, open the dialog and try to move the draggable elements. Check if the element could be dropped. Use this application as a starting point:

https://stackblitz.com/edit/github-ilmjhj?file=src%2Fapp%2Fdialog-sample-1%2Fdialog-sample-1.component.ts